src/main/java/org/jitsi/jicofo/xmpp/FocusComponent.java

```diff
@@ -80,67 +83,60 @@
      */
     private ReservationSystem reservationSystem;
 
+    private final Connector connector = new Connector();
+
     /**
      * Creates new instance of <tt>FocusComponent</tt>.
-     * @param config Contains the configuration for the XMPP component connection.
-     * @param anonymousFocus indicates if the focus user is anonymous.
-     * @param focusAuthJid the JID of authenticated focus user which will be
-     *                     advertised to conference participants.
      */
-    public FocusComponent(XmppComponentConfig config, boolean anonymousFocus, String focusAuthJid)
+    public FocusComponent(XmppComponentConfig config, boolean isFocusAnonymous, String focusAuthJid)
     {
         super(config.getHostname(), config.getPort(), config.getDomain(), config.getSubdomain(), config.getSecret());
 
-        this.isFocusAnonymous = anonymousFocus;
+        this.isFocusAnonymous = isFocusAnonymous;
         this.focusAuthJid = focusAuthJid;
     }
 
-    /**
-     * Initializes this component.
-     */
-    public void init()
+    public void setFocusManager(FocusManager focusManager)
     {
-        OSGi.start(this);
+        this.focusManager = focusManager;
     }
 
-    /**
-     * Method will be called by OSGi after {@link #init()} is called.
-     */
-    @Override
-    public void start(BundleContext bc)
-        throws Exception
+    public void setAuthAuthority(AuthenticationAuthority authAuthority)
     {
-        ConfigurationService configService
-            = ServiceUtils2.getService(bc, ConfigurationService.class);
+        this.authAuthority = authAuthority;
+    }
 
-        loadConfig(configService, "org.jitsi.jicofo");
+    public void setReservationSystem(ReservationSystem reservationSystem)
+    {
+        this.reservationSystem = reservationSystem;
+    }
 
-        if (!isPingTaskStarted())
-            startPingTask();
 
-        authAuthority = ServiceUtils2.getService(bc, AuthenticationAuthority.class);
-        focusManager = ServiceUtils2.getService(bc, FocusManager.class);
-        reservationSystem = ServiceUtils2.getService(bc, ReservationSystem.class);
+    public void loadConfig(ConfigurationService config, String configPropertiesBase)
+    {
+        super.loadConfig(config, configPropertiesBase);
     }
 
-    /**
-     * Releases resources used by this instance.
-     */
-    public void dispose()
+    public void connect()
     {
-        OSGi.stop(this);
+        if (!isPingTaskStarted())
+        {
+            startPingTask();
```

Comment:
Does this need to be atomic with the `isPingTaskStarted()` check?

Reply:
In theory -- yes. In practice -- no. That's the only place any of that code is used, it [has always been](https://github.com/jitsi/jicofo/blob/master/src/main/java/org/jitsi/jicofo/xmpp/FocusComponent.java#L118) like this, and I'm hoping to remove all that code soon, so I'd rather not add to it.

Reply:
Cool, just checking.